### PR TITLE
Fixes #4173 - Show 'edited by' Messages

### LINKED ISSUE #4173: Show 'edited by' Messages - moderators are not affected
https://community.mybb.com/thread-230145-post-1358450.html#pid1358450

### PR DESCRIPTION
**Pull request by Eldenroot**
Resolves #4173 

reported+fix by @Supryk

### CODE REVIEW

File: inc/functions_post.php [hunk 1]
```diff
@@ -551,7 +551,7 @@ function build_postbit($post, $post_type=0)
 		}
 
 		// Figure out if we need to show an "edited by" message
-		if($post['edituid'] != 0 && $post['edittime'] != 0 && $post['editusername'] != "" && (($mybb->settings['showeditedby'] != 0 && $usergroup['cancp'] == 0) || ($mybb->settings['showeditedbyadmin'] != 0 && $usergroup['cancp'] == 1)))
+		if($post['edituid'] != 0 && $post['edittime'] != 0 && $post['editusername'] != "" && (($mybb->settings['showeditedby'] != 0 && $usergroup['cancp'] == 0) || ($mybb->settings['showeditedbyadmin'] != 0 && $usergroup['cancp'] == || is_moderator($post['fid'], "", $post['uid'])))))
```
Comment (euantorano) [inline inc/functions_post.php]: This is a syntax error, specifically:

```
$usergroup['cancp'] == || is_moderator($post['fid'], "", $post['uid']))
```

I guess the below suggestion is correct but haven't tested the actual logic:

```suggestion
		if($post['edituid'] != 0 && $post['edittime'] != 0 && $post['editusername'] != "" && (($mybb->settings['showeditedby'] != 0 && $usergroup['cancp'] == 0) || ($mybb->settings['showeditedbyadmin'] != 0 && $usergroup['cancp'] == 1 || is_moderator($post['fid'], "", $post['uid'])))))
```

**Review by euantorano — CHANGES_REQUESTED** [review]

**Comment by Eldenroot** [comment]
Ok, thx!

**Comment by Sama34** [comment]
Could you remove the ` && $usergroup['cancp'] == 0` check ?

As per the setting description;
> Show 'edited by' Messages
> Once a post is edited by a regular user, do you want to show the edited by message?

Why are users with ACP access excepted of this _supposed_ general setting? Hope my inquiry is clear. @euantorano

**Comment by Eldenroot** [comment]
@Sama34 

```
Show 'edited by' Message for Administrators and Moderators
Do you want to show edited by messages for posts created by administrators and moderators?
```

**Comment by Eldenroot** [comment]
@euantorano @Sama34 - OK, there is no issue to made the required change but please make a final decision :)

**Comment by Sama34** [comment]
I think I got confused. Instead, you should add the `is_moderator` check to the previous conditional as well. Like so:
`$mybb->settings['showeditedby'] != 0 && $usergroup['cancp'] == 0 && !is_moderator($post['fid'], "", $post['uid'])`

Ignore my previous inquiry.

**Comment by Eldenroot** [comment]
@Sama34 @euantorano - so final decision? This should be fine :)

**Comment by dvz** [comment]
The added condition should be related to `$mybb->settings['showeditedbyadmin'] != 0`, but may evaluate as true independently when `$mybb->settings['showeditedbyadmin'] != 0 && $usergroup['cancp'] == 1` evaluates as false (missing group parenthesis).
@Sama34's change is also needed - otherwise notices related to users without ACP access (`$usergroup['cancp'] == 0`), but with moderator permissions, will still be shown, even with `showeditedbyadmin` setting enabled.

**Comment by Sama34** [comment]
So the following might be the desired result:
```
if(
	$post['edituid'] != 0 && $post['edittime'] != 0 && $post['editusername'] != "" && (
		($mybb->settings['showeditedby'] != 0 && $usergroup['cancp'] == 0 && !is_moderator($post['fid'], "", $post['uid'])) ||
		($mybb->settings['showeditedbyadmin'] != 0 && (
			$usergroup['cancp'] == 1 || is_moderator($post['fid'], "", $post['uid'])
		))
	)
)
```

Modified for improved readability. @Eldenroot

**Pull request merged**